src/execution/operator/join/physical_hash_join.cpp

```diff
@@ -191,6 +182,12 @@ void PhysicalHashJoin::GetChunkInternal(ExecutionContext &context, DataChunk &ch
 		// empty hash table with INNER, RIGHT or SEMI join means empty result set
 		return;
 	}
+	// In case we have a perfect hash table, probe it
+	if (pjoin_executor && use_perfect_hash) {
```

Comment:
Can we check only `use_perfect_hash` here and assert the `pjoin_executor` exists if that is true?